CONTRIBUTING.md

```diff
@@ -0,0 +1,200 @@
+
+### Step 2. Make code changes
+
+To make code changes, you need to fork the repository. You will need to setup a
+development environment and run the unit tests. This is covered in the section
+"Setup environment".
+
+### Step 3. Create a pull request
+
+Once the change is ready, open a pull request from your branch in your fork to
+the master branch in [keras-team/keras](https://github.com/keras-team/keras).
+
+### Step 4. Sign the Contributor License Agreement
+
+After creating the pull request, the `cla/google` check will be performed and,
+if you haven't signed the Contributor License Agreement (CLA), it will fail with
+instructions on how to do so. Please follow the instructions to sign the CLA and
+the check will pass.
+
+![CLA signed](https://github.com/keras-team/keras/assets/1091026/71c26353-e3b5-4135-8bae-64693c717775)
+
+### Step 5. Code review
+
+If the tests fail, look into the error messages and try to fix them.
+
+![CI tests](https://github.com/keras-team/keras/assets/1091026/6f6c17ef-6bd7-4e95-9fbc-1906cde37380)
+
+A reviewer will review the pull request and provide comments. There may be
+several rounds of comments and code changes before the pull request gets
+approved by the reviewer.
+
+![Approval from reviewer](https://github.com/keras-team/keras/assets/1091026/8d28f74c-21e9-4146-b0ff-62d649a552a8)
+
+### Step 6. Merging
+
+Once the pull request is approved, a `ready to pull` tag will be added to the
+pull request. A team member will take care of the merging.
+
+![Ready to pull and merged](https://github.com/keras-team/keras/assets/1091026/c3908345-d7ae-44ee-a428-01f3b448b46b)
+
+Here is an [example pull request](https://github.com/keras-team/keras/pull/18848)
+for your reference.
+
+## Setup environment
+
+We provide two ways of setting up a development environment. One is to use a
+dev container, and the other one is to setup a local environment by installing
+the dev tools needed.
+
+### Option 1: Use a Dev container
+
+We provide multiple dev container options. Please read the
+[Dev container documentation](https://github.com/keras-team/keras/tree/master/.devcontainer).
+
+### Option 2: Setup a local environment
+
+To setup your local dev environment, you will need the following tools.
+
+1.  [git](https://github.com/) for code repository management.
+2.  [python](https://www.python.org/) to build and code in Keras.
+
+The following commands check the tools above are successfully installed. Note
+that Keras requires at least Python 3.9 to run.
+
+```shell
+git --version
+python --version
+```
+
+Clone your forked repo to your local machine. Go to the cloned directory to
+install the dependencies into the venv. Since `tf-nightly` uses `keras-nightly`
+as a dependency, we need to uninstall `keras-nightly` so that tests will run
+against Keras code in the local workspace.
+
+```shell
+git clone https://github.com/YOUR_GITHUB_USERNAME/keras.git
+cd keras
+pip install -r requirements.txt
+pip uninstall keras-nightly
```

Comment:
Is this really necessary?

Reply:
This question refers to this line only, right? `pip uninstall keras-nightly`

(github always shows 4 lines and doesn't show what was highlighted)

Reply:
Yes, why uninstall? if you run commands from the base dir it will pick up the local `keras` dir.

Reply:
You're right. Expect maybe in some extreme case where a local sub-directory is removed.

Anyway, I'll remove this line and the blurb about it above.